streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1938,115 +1938,134 @@ object ZChannel {
-                               for {
-                                 _ <- permits
-                                        .withPermit(latch.succeed(()) *> raceIOs)
-                                        .interruptible
-                                        .forkIn(childScope)
-                                 _ <- latch.await
-                               } yield ()
-                             }
-                         case MergeStrategy.BufferSliding =>
-                           (fiberId: FiberId) =>
-                             pull.flatMap { channel =>
-                               val canceler = Promise.unsafe.make[Nothing, Unit](fiberId)(Unsafe.unsafe)
-                               val latch    = Promise.unsafe.make[Nothing, Unit](fiberId)(Unsafe.unsafe)
-                               for {
-                                 size <- cancelers.size
-                                 _    <- ZIO.when(size >= n)(cancelers.take.flatMap(_.succeed(())))
-                                 _    <- cancelers.offer(canceler)
-                                 raceIOs =
-                                   ZIO.scopedWith { scope =>
-                                     (incoming >>> channel)
-                                       .toPullInAlt(scope)
-                                       .flatMap(evaluatePull(_).race(canceler.await.interruptible))
-                                   }
-                                 _ <- permits
-                                        .withPermit(latch.succeed(()) *> raceIOs)
-                                        .interruptible
-                                        .forkIn(childScope)
-                                 _ <- latch.await
-                               } yield ()
-                             }
-                       }
-        _ <- pullStrategy(fiberId).forever
-               .catchAllCause(cause =>
-                 cause.failureOrCause match {
-                   case Left(_: Left[OutErr, OutDone]) => outgoing.offer(Exit.failCause(cause))
-                   case Left(x: Right[OutErr, OutDone]) =>
-                     permits.withPermits(n.toLong)(ZIO.unit).interruptible *>
-                       lastDone.get.flatMap {
-                         case Some(lastDone) => outgoing.offer(Exit.fail(Right(f(lastDone, x.value))))
-                         case None           => outgoing.offer(Exit.fail(Right(x.value)))
-                       }
-                   case Right(cause) => outgoing.offer(Exit.failCause(cause.map(Left(_))))
-                 }
-               )
-               .raceFirst(awaitErrorSignal(childScope, fiberId)(errorSignal))
-               .forkIn(scope)
-      } yield {
-        lazy val consumer: ZChannel[Env, Any, Any, Any, OutErr, OutElem, OutDone] =
-          unwrap[Env, Any, Any, Any, OutErr, OutElem, OutDone] {
-            outgoing.take.flatten.foldCause(
-              cause =>
+      ZIO.fiberIdWith { fiberId =>
+        ZIO.suspendSucceed {
```

Comment:
You don't need to suspend here, `ZIO.fiberIdWith` already takes care of that

Reply:
Done